src/map/script.cpp

```diff
@@ -11337,18 +11337,28 @@ BUILDIN_FUNC(sc_end)
 
 /**
  * Ends all status effects from any learned skill on the attached player.
- * sc_end_class {<char_id>};
+ * if <job_id> was given it will end all the effect for that job
+ * if <job_id> is 0 it will return the job of the invoking character
+ * sc_end_class {<job_id>{,<char_id>}};
  */
 BUILDIN_FUNC(sc_end_class)
 {
 	struct map_session_data *sd;
 	uint16 skill_id;
-	int i;
+	int class_ = 0;
+	struct block_list *bl = map_id2bl(st->rid);
 
-	if (!script_charid2sd(2, sd))
-		return SCRIPT_CMD_FAILURE;
+	if (!script_charid2sd(3, sd)) {
+		script_pushint(st, -1);
```

Comment:
Please keep this as SCRIPT_CMD_FAILURE and remove the value push.
This is intended for future merge of #2063